internal/executor/terminalwrapper/terminalwrapper.go

```diff
@@ -46,6 +51,25 @@ func New(
 			Locator:            locator,
 			TrustedSecret:      trustedSecret,
 		})
+		if err != nil {
+			return err
+		}
+
+		if !wrapper.lifecycleStartedSent {
```

Comment:
The client will refresh automatically. I'm not sure if we need to bother the user with that 🤔

Reply:
I guess it's hard to see the need of sending the started event only once without the backend/frontend code that handles it. That's why `lifecycleStartedSent` field is so bothering me. IMO looks a bit too over thinking at the moment.

Reply:
>I guess it's hard to see the need of sending the started event only once without the backend/frontend code that handles it.

I thought we would just do minor conversions at the backend and bounce the lifecycle event to the web UI, without de-duplicating it.

`lifecycleStartedSent` does de-duplication at the agent's side, which makes some sense, because the agent already stores state.

Do you want to move the de-duplication out of the agent?

Reply:
I just don't see a need for the de-duplication. In case of a good network and when everything goes smooth there will be no need for the de-duplication. And it's hopefully %99.9+ of the time. If there are some issues though, IMO it might be OK to get several notifications. Service worker in this case can try to force fetch task again to reconnect or do some other stuff.

Reply:
Removed de-duplication in https://github.com/cirruslabs/cirrus-ci-agent/pull/192/commits/4d565dc3624fd5328bbb51b3f4c5a44dddacbe88.